### PR TITLE
feature:database - add rawQuery operation on write_inferface

### PR DESCRIPTION
**Pull request by iancardosozup**
**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by matheusalcantarazup** [comment]
I think that this PR depends on #159?

File: pkg/services/database/database.go [hunk 1]
```diff
@@ -243,3 +243,21 @@ func (d *database) findPreloadWitLimitAndPageQuery(
...
+		return err
+	}
+
+	tx, err := sqlDB.Begin()
+	if err != nil {
+		return err
+	}
+
+	if _, err := tx.Exec(rawQuery); err != nil {
+		return err
```
Comment (matheusalcantarazup) [inline pkg/services/database/database.go:259]: We shouldn't call tx.Rollback() before return?

File: pkg/services/database/database.go [hunk 2]
```diff
@@ -243,3 +243,21 @@ func (d *database) findPreloadWitLimitAndPageQuery(
 
 	return d.connectionRead.Table(table).Where(where).Limit(limit).Offset(page * limit)
 }
+
+func (d *database) Batch(rawQuery string) error {
```
Comment (matheusalcantarazup) [inline pkg/services/database/database.go:247]: Can we add a documentation about this method?

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]
Its not clear to me witch batch operation is happening here. To me the Batch() method is just executing a raw query. I'm missing something here?

**Comment by iancardosozup** [comment]
> I think that this PR depends on #159?

yeah, forgot to mention

File: pkg/services/database/database.go (hunk 2 above)
Comment (wiliansilvazup) [inline pkg/services/database/database.go]: Why can't this method keep the default of the others by returning `response.IResponse` ?

File: pkg/services/database/database.go [hunk 3]
```diff
@@ -243,3 +243,23 @@ func (d *database) findPreloadWitLimitAndPageQuery(
 
 	return d.connectionRead.Table(table).Where(where).Limit(limit).Offset(page * limit)
 }
+
+// Deprecated: RawQuery starts a transaction and try to execute the raw query into database.
+// is not recommended using this and the method will not be available after cli v2.10.0
+func (d *database) RawQuery(rawQuery string) error {
```
Comment (wiliansilvazup) [inline pkg/services/database/database.go:249]: [Keep default format equals read interface](https://github.com/ZupIT/horusec-devkit/blob/cb1165154a4c15a69284a3cba604cbd31f4ca7ca/pkg/services/database/read_interface.go#L27)

In Write case entityPointer isn't necessary:
```suggestion
func (d *database) RawWrite(rawSQL string, values ...interface{}) response.IResponse {
```

And in the future we can change Read too:
```
func (d *database) RawRead(rawSQL string, entityPointer interface{}, values ...interface{})
```

File: pkg/services/database/database.go [hunk 4]
```diff
@@ -243,3 +243,23 @@ func (d *database) findPreloadWitLimitAndPageQuery(
...
+	sqlDB, err := d.connectionWrite.DB()
+	if err != nil {
+		return err
+	}
+
+	tx, err := sqlDB.Begin()
+	if err != nil {
+		return err
+	}
```
Comment (wiliansilvazup) [inline pkg/services/database/database.go:258]: Why you can use `StartTransaction` method?

**Review by wiliansilvazup — CHANGES_REQUESTED** [review]

File: pkg/services/database/database_test.go [hunk 5]
```diff
@@ -821,3 +821,93 @@ func TestFindPreloadWitLimitAndPage(t *testing.T) {
...
+		err = database.RawQuery(query)
+
+		assert.Error(t, err)
+	})
+	t.Run("should error when get DB fails", func(t *testing.T) {
+		db := &gorm.DB{
+			Config: &gorm.Config{},
+		}
+
+		query := "UPDATE table a SET a.value = 10 where a.id = 123 ;"
```
Comment (wiliansilvazup) [inline pkg/services/database/database_test.go:902]: It is a security flaw to pass values in this format see the [GORM documentation](https://gorm.io/docs/sql_builder.html#Raw-SQL) to understand more about it.

**Review by wiliansilvazup — CHANGES_REQUESTED** [review]

File: pkg/services/database/database.go (hunk 4 above)
Comment (iancardosozup) [reply pkg/services/database/database.go:258]: we cannot access the pure sql object with StartTransaction, to be able to execute a query with more than one statement i couldn't find another way, i welocme suggestions

File: pkg/services/database/database_test.go (hunk 5 above)
Comment (iancardosozup) [reply pkg/services/database/database_test.go:902]: this is just a sample query that raw query would use, and the function in particular don't use GORM standards or it's functions

File: pkg/services/database/database.go (hunk 4 above)
Comment (wiliansilvazup) [reply pkg/services/database/database.go:258]: okay, we can keep in this format

**Review by matheusalcantarazup — CHANGES_REQUESTED** [review]

**Comment by nathanmartinszup** [comment]
Rebase branch with main to fix conflicts and simplify the review.

**Review by nathanmartinszup — APPROVED** [review]

**Pull request merged**